src/main/java/com/microsoft/graph/requests/upload/UploadSliceRequestBuilder.java

```diff
@@ -0,0 +1,79 @@
+package com.microsoft.graph.requests.upload;
+
+import com.microsoft.graph.models.UploadResult;
+import com.microsoft.kiota.*;
+import com.microsoft.kiota.serialization.Parsable;
+import com.microsoft.kiota.serialization.ParsableFactory;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import okhttp3.Response;
+
+import javax.annotation.Nonnull;
+import java.io.InputStream;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+
+public class UploadSliceRequestBuilder<T extends Parsable> {
+
+    private final UploadResponseHandler responseHandler;
+    private final RequestAdapter requestAdapter;
+    private final String urlTemplate;
+
+    private final long rangeBegin;
+    private final long rangeEnd;
+    private final long totalSessionLength;
+
+    private final int rangeLength;
+
+    private final ParsableFactory<T> factory;
```

Comment:
I'm not sure if you need this property. As you already have the type `T`, you can eliminate this and pass a lambda/function that simply returns a new instance of `T` to the response handler. 
This will reduce the number of parameters that the consumer needs to pass to init the task.